src/modules/ssh_apt/entity/index.ts

```diff
@@ -0,0 +1,92 @@
+import { Column, Entity, JoinColumn, ManyToOne, PrimaryGeneratedColumn, Unique } from 'typeorm';
+
+import { cloudId } from '../../../services/cloud-id';
+import { SshCredentials } from '../../ssh_accounts/entity';
+
+@Entity()
+@Unique('uq_package_server_package_version_arch', ['server', 'package', 'version', 'architecture'])
+export class Package {
+  /**
+   * @private
+   * Auto-incremented ID field
+   */
+  @PrimaryGeneratedColumn()
+  id?: number;
+
+  /**
+   * @public
+   * Name of the ssh server
+   */
+  @Column({
+    type: 'varchar',
+    nullable: false,
+  })
+  @ManyToOne(() => SshCredentials, { nullable: false })
+  @JoinColumn({ referencedColumnName: 'name', name: 'server' })
+  @cloudId
+  server: string;
+
+  /**
+   * @public
+   * Name of the package
+   */
+  @Column({
+    type: 'varchar',
+    nullable: false,
+  })
+  @cloudId
+  package: string;
+
+  /**
+   * @public
+   * The version of the package
+   */
+  @Column({
+    type: 'varchar',
+    nullable: false,
+  })
+  @cloudId
+  version: string;
+
+  /**
+   * The architecture of the package
+   */
+  @Column({
+    type: 'varchar', // TODO: Maybe make this an enum?
+    nullable: false,
+  })
+  @cloudId
+  architecture: string;
+
+  /**
+   * @public
+   * The description of the package
+   */
+  @Column({
+    type: 'varchar',
+    nullable: true,
+  })
+  description?: string;
+
+  /**
+   * @public
+   * Whether or not the package is installed
+   */
+  @Column({
+    type: 'boolean',
+    nullable: false,
+    default: false,
+  })
+  installed: boolean;
```

Comment:
So you're keeping all the "possible" packages here and setting `installed: true` for those who are going to be/are installed?

Reply:
Why not just keeping the installed packages? And `INSERT` when we're going to install a new package?

Reply:
So there are several reasons why I went with this approach:

1. You can still see just the set that are installed with a `SELECT * FROM package WHERE installed = TRUE`
2. If you remember some phrase that describes the package you want to install but can't remember the exact package name, you can `SELECT * FROM package WHERE description LIKE '%sound editor%' AND installed = FALSE` and then look for the actual package you want to install.
3. If you can't remember the exact spelling of the package, you can use one of the [fuzzy string matching functions](https://www.postgresql.org/docs/current/fuzzystrmatch.html) to help you out, like `SELECT *, levenshtein_less_equal(package, "emacks", 5) AS score FROM package WHERE score < 5 ORDER BY score`
4. We could potentially put a [full-text search index](https://www.postgresql.org/docs/current/textsearch-intro.html#TEXTSEARCH-MATCHING) on the package description to allow easier Google-style searching with minimal symbols in the search text (though this feature is aimed at developers exposing that to end users and still requires a lot of ceremony in the query itself; I also didn't add it to the PR (yet?) because I didn't want to conflate potential sources for the current performance issues by using a new index type)
5. By listing all packages, you can also see which versions are available for installation with `SELECT version FROM package WHERE package = 'vim'`

Basically, there's a lot of classic database value :tm: that you get when the table is fully populated in this way that you would lose if it only listed the currently installed packages. What packages can I install? I dunno! What versions are available? I dunno! Does my CPU architecture have a special version of it? I dunno!

On top of that, Ubuntu/Debian VMs on AWS are pretty "fat" by default. With almost nothing extra installed (just Docker), there's over 8k packages installed on this machine, so the performance issues would probably still rear their head here and we should tackle that, anyways.